src/theory/quantifiers/sygus/sygus_reconstruct.h

```diff
@@ -0,0 +1,340 @@
+   * @param sol constant solution to `ob`
+   */
+  void markSolved(Node k, Node s);
+
+  /**
+   * Initialize a sygus enumerator and a candidate rewrite database for each
+   * sygus datatype type.
+   *
+   * @param stn The sygus datatype type encoding the syntax
+   * restrictions
+   */
+  void initializeEnumeratorsAndDatabases(TypeNode stn);
+
+  /**
+   * Returns the next enumerated term for the given sygus datatype type.
+   *
+   * @param stn The sygus datatype type to enumerate a term for
+   * @return The enumerated sygus term
+   */
+  Node nextEnum(TypeNode stn);
+
+  /**
+   * Remove solved obligations from `d_unsolvedObs.
+   */
+  void removeSolvedObs();
+
+  /**
+   * Replace all bound variables in `n` with ground values of the type.
+   *
+   * @param n A term containing bound variables
+   * @return `n` with all vars in `n` replaced with ground values
+   */
+  Node replaceVarsWithGroundValues(Node n) const;
+
+  /**
+   * A notification that s is equal to n * { vars -> subs }. This function
+   * should return false if we do not wish to be notified of further matches.
+   */
+  bool notify(Node s,
+              Node n,
+              std::vector<Node>& vars,
+              std::vector<Node>& subs) override;
+
+  /**
+   * Reset the state of this SygusReconstruct object
+   */
+  void reset();
+
+  /**
+   * Return a string representation of an obligation.
+   *
+   * @param k An obligation
+   * @return A string representation of `k`
+   */
+  std::string ob(Node k) const;
+
+  /**
+   * Print all reachable obligations and their candidate solutions from
+   * the `root` obligation and its candidate solutions.
+   *
+   * \note requires enabling "sygus-rcons" trace
+   *
+   * @param root The root obligation to start from
+   */
+  void printCandSols(const Node& mainOb) const;
+
+  /**
+   * Print the pool of patterns/shape used in the matching phase.
+   *
+   * \note requires enabling "sygus-rcons" trace
+   *
+   * @param pool a pool of patterns/shapes to print
+   */
+  void printPool(const std::unordered_map<TypeNode,
+                                          std::vector<Node>,
+                                          TypeNodeHashFunction>& pool) const;
+
+  /** pointer to the sygus term database */
+  TermDbSygus* d_tds;
+  /** reference to the statistics of parent */
+  SygusStatistics& d_stats;
+
+  /** a set of obligations that are not yet satisfied for each sygus datatype */
+  std::unordered_map<TypeNode,
+                     std::unordered_set<Node, NodeHashFunction>,
+                     TypeNodeHashFunction>
+      d_unsolvedObs;
+  /** a map from an obligation to its info */
+  std::unordered_map<Node, ObligationInfo, NodeHashFunction> d_info;
+  /** a map from a builtin term to its obligation. The sygus type of the
+   * obligation is needed as different sygus types may have obligations to
+   * reconstruct the same builtin term */
+  std::unordered_map<TypeNode,
```

Comment:
Similar to obligation info, can we consolidate a map for SygusTypeRConsInfo?  I believe d_unsolvedObs, d_ob, d_crds, d_enumerators could be grouped into it.

Reply:
Done for `d_ob`, `d_crds`, and `d_enumerator`. `d_unsolvedObs` is now local to `SygusReconstruct::reconstructSolution` method.